routes/drones.js

```diff
@@ -1,36 +1,105 @@
 const express = require('express');
 const router = express.Router();
 
-// require the Drone model here
+// require the Drone model 
+
+const Drone = require("../models/Drone.model");
 
 router.get('/drones', (req, res, next) => {
   // Iteration #2: List the drones
-  // ... your code here
+  Drone.find()
+    .then( (dronesFromDB) => {
+      const data = {
+        dronesArr: dronesFromDB
+      };
+      res.render("drones/list", data)
+    })
+    .catch( (error) => {
+      console.log("Error getting data from DB", error);
+      next(error);
+    })
 });
 
 router.get('/drones/create', (req, res, next) => {
   // Iteration #3: Add a new drone
-  // ... your code here
+  router.get("/drones/create", (req, res) => {
+    res.render("drones/drone-create");
+  })
 });
 
 router.post('/drones/create', (req, res, next) => {
   // Iteration #3: Add a new drone
-  // ... your code here
+
+  const drones = {
+    name: req.body.name,
+    propellers: req.body.propellers,
+    maxSpeed: req.body.MaxSpeed,
+  };
+
+  Drone.create(drones)
+    .then( () => {
+      res.redirect("/drones");
+    })
+    .catch( (error) => {
+      console.log("Error creating drone in the DB", error);
+      next(error);
+    })
 });
 
 router.get('/drones/:id/edit', (req, res, next) => {
   // Iteration #4: Update the drone
-  // ... your code here
+  const {droneId} = req.params;
+
+  Drone.findById(droneId)
+    .then( (drones) => {
+      res.render("drones/62cc51cd565a994b2f9d05e7/edit", drones);
+    })
+    .catch( (error) => {
+      console.log("Error getting drone details from DB", error);
+      next(error);
+    })
+
+ 
 });
 
 router.post('/drones/:id/edit', (req, res, next) => {
   // Iteration #4: Update the drone
-  // ... your code here
+
+  const droneId = req.params.droneId;
+
+  const newDetails = {
+    name: req.body.name,
+    propellers: req.body.propellers,
+    maxSpeed: req.body.maxSpeed,
```

Comment:
I'll point here that this =>  `maxSpeed`  is not the `name` on the `create-form.hbs` => you used `max-speed` so it should have been  `maxSpeed: req.body.max-speed`  and it should be the same along the your code, at some place its `MaxSpeed` and in another one is `maxSpeed`